doc/X-OR_tutorial/X-OR_tutorial.rst

```diff
@@ -590,20 +590,20 @@ To do so we are going to add the following to our pass:
         1. Transform `b`, `c` and `d` into `b'`, `c'` and `d'`
         2. Create `add1'` such as `add1' = b' + c'`
         3. Create `add2'` such as `add2' = add1' + d'`
-        4. Transform `add2'` into its base 2 equivalent `add2`
+        4. Apply modulo 2 on `add2'` bits and transform into base 2 gives us `add2`
         5. Store `add2` in `a`
 
      This will reduce the number of transformations, which will reduce the number of instructions generated, making the code faster and the obfuscation a little less obvious.
-     This not as trivial as this but we will get the details sorted out later.
+     This is not that trivial, but we will get the details sorted out later.
 
     * If you have taken a look at the non-optimized obfuscated code, you've probably noticed that the pattern is very easy to spot.
-      There a nice exponential drawn by the increasing length of the base successive exponents.
+      There's a nice exponential drawn by the increasing length of the base successive exponents.
```

Comment:
when computing the base change, you're computing the successive exponents of the base.